e2e-tests/src/driver.rs

```diff
@@ -97,131 +97,113 @@ impl Driver {
         let mut report = TestRunReport::new();
         let mut initialized_protocols: HashMap<String, ProtocolSandboxEnvironment> = HashMap::new();
 
-        // Run scenarios directory by directory
-        let scenarios_dir = self.environment.input_dir.join("scenarios");
-        let mut entries = read_dir(scenarios_dir).await?;
-
-        while let Some(entry) = entries.next_entry().await? {
-            let path = entry.path();
-            if path.is_dir() {
-                let test_file_path = path.join("test.json");
+        let protocols_dir = self.environment.input_dir.join("protocols");
+
+        for protocol_name in &self.environment.protocols {
+            let protocol_name = protocol_name.to_string().to_lowercase();
+
+            if initialized_protocols.contains_key(&protocol_name) {
+                continue;
+            }
+
+            for sandbox_cfg in &self.config.protocol_sandboxes {
+                let config_protocol_name = match sandbox_cfg {
+                    ProtocolSandboxConfig::Stellar(_) => "stellar",
+                    ProtocolSandboxConfig::Near(_) => "near",
+                    ProtocolSandboxConfig::Icp(_) => "icp",
+                    ProtocolSandboxConfig::Ethereum(_) => "ethereum",
+                };
+
+                if config_protocol_name == protocol_name {
+                    let sandbox_env = match sandbox_cfg {
+                        ProtocolSandboxConfig::Stellar(config) => {
+                            ProtocolSandboxEnvironment::Stellar(StellarSandboxEnvironment::init(
+                                config.clone(),
+                            )?)
+                        }
+                        ProtocolSandboxConfig::Near(config) => ProtocolSandboxEnvironment::Near(
+                            NearSandboxEnvironment::init(config.clone()).await?,
+                        ),
+                        ProtocolSandboxConfig::Icp(config) => ProtocolSandboxEnvironment::Icp(
+                            IcpSandboxEnvironment::init(config.clone())?,
+                        ),
+                        ProtocolSandboxConfig::Ethereum(config) => {
+                            ProtocolSandboxEnvironment::Ethereum(EthereumSandboxEnvironment::init(
+                                config.clone(),
+                            )?)
+                        }
+                    };
+
+                    initialized_protocols.insert(protocol_name.clone(), sandbox_env);
+                    break;
+                }
+            }
+        }
+
+        for protocol_name in &self.environment.protocols {
+            let protocol_name = protocol_name.to_string().to_lowercase();
+            let protocol_path = protocols_dir.join(&protocol_name);
+
+            if !protocol_path.exists() || !protocol_path.is_dir() {
```

Comment:
[`Utf8Path::is_dir`](https://docs.rs/camino/latest/camino/struct.Utf8PathBuf.html#method.is_dir) will only be true if it exists and is a dir

```suggestion
            if !protocol_path.is_dir() {
```